server/lib/scheduler/RootParentFinder.ts

```diff
@@ -0,0 +1,47 @@
+import * as models from "../models";
```

Comment:
Should this be in the scheduler? It feels very general purpose to me, I'm surprised it wasn't needed earlier. Seems like it should be a method on ``workspace`` (``#rootParent``)